packages/core/src/engines/query/RouterQueryEngine.ts

```diff
@@ -55,7 +55,7 @@ async function combineResponses(
  * A query engine that uses multiple query engines and selects the best one.
  */
 export class RouterQueryEngine extends PromptMixin implements BaseQueryEngine {
-  serviceContext: ServiceContext;
+  serviceContext?: ServiceContext;
```

Comment:
how about removing the serviceContext from classes (but keeping it from the init parameter)?

my guess is that users are never reading the serviceContext and if they are using it, they are passing it as init parameter (that's what all our example code does)